integ-test/src/test/java/org/opensearch/sql/ppl/SortCommandIT.java

```diff
@@ -38,17 +44,62 @@ public void testSortWithNullValue() throws IOException {
             String.format(
                 "source=%s | sort balance | fields firstname, balance",
                 TEST_INDEX_BANK_WITH_NULL_VALUES));
+
+    JSONArray dataRows = result.getJSONArray("datarows");
+    List<Object[]> nullRows = new ArrayList<>();
+    for (int i = 0; i < dataRows.length(); i++) {
+      JSONArray row = dataRows.getJSONArray(i);
+      if (row.isNull(1)) {
+        nullRows.add(new Object[] {row.getString(0), null});
+      }
+    }
+    // Verify the set values for null balances as rows with null balance can return in any order
+    List<Object[]> expectedNullRows =
+        Arrays.asList(
+            new Object[] {"Hattie", null},
+            new Object[] {"Elinor", null},
+            new Object[] {"Virginia", null});
+    assertSetEquals(expectedNullRows, nullRows);
+
+    // Create a new JSONArray to store not null balance rows
+    JSONArray nonNullBalanceRows = new JSONArray();
+
+    for (int i = 0; i < dataRows.length(); i++) {
+      JSONArray row = dataRows.getJSONArray(i);
+      if (!row.isNull(1)) { // Check if the balance (index 1) is not null
+        nonNullBalanceRows.put(row);
+      }
+    }
+
+    // Create a new JSONObject with the filtered data
+    JSONObject filteredResult = new JSONObject();
+    filteredResult.put("schema", result.getJSONArray("schema"));
+    filteredResult.put("total", nonNullBalanceRows.length()); // Update total count
+    filteredResult.put("datarows", nonNullBalanceRows);
+    filteredResult.put("size", nonNullBalanceRows.length()); // Update size count
+
```

Comment:
It's becoming very long. Please consider extraction for readability. Even if we use the extracted method only once, it is better extracting for readability.

Reply:
Updated